aten/src/ATen/native/cpu/PowKernel.cpp

```diff
@@ -34,126 +36,104 @@ void pow_tensor_tensor_kernel(TensorIteratorBase& iter) {
   }
 }
 
-void pow_tensor_scalar_kernel(TensorIteratorBase& iter, const Scalar& exp_scalar) {
-  if (isFloatingType(iter.dtype())) {
-    const auto exp = exp_scalar.to<double>();
-    // Floating types allow AVX2 vector optimizations for pow/sqrt/rsqrt:
-    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "pow", [&]() {
+// Floating and complex types (minus Half) allow for optimized kernels
+template <typename scalar_t, typename cast_scalar_t, typename exp_scalar_t>
+void pow_tensor_scalar_optimized_kernel(TensorIteratorBase& iter, const exp_scalar_t exp) {
+  using Vec = Vec256<scalar_t>;
+  if (exp == 0.5) {
+    cpu_kernel_vec(iter,
+        [](scalar_t base) -> scalar_t {
+        	return std::sqrt(base);
+        },
+        [](Vec base) -> Vec { return base.sqrt(); }
+    );
+  } else if (exp == 2.0) {
+    cpu_kernel_vec(iter,
+        [](scalar_t base) -> scalar_t {
+        	return base * base;
+        },
+        [](Vec base) -> Vec { return base * base; }
+    );
+  } else if (exp == 3.0) {
+    cpu_kernel_vec(iter,
+        [](scalar_t base) -> scalar_t {
+        	return base * base * base;
+        },
+        [](Vec base) -> Vec { return base * base * base; }
+    );
+  } else if (exp == -0.5) {
+    cpu_kernel_vec(iter,
+        [](scalar_t base) __ubsan_ignore_float_divide_by_zero__ -> scalar_t {
+          return scalar_t(1.0) / std::sqrt(base);
```

Comment:
also update these to `static_cast<cast_scalar_t>`

Reply:
Oops, missed these.

Reply:
I think these ones should be `static_cast<scalar_t>` instead.
`static_cast<cast_scalar_t>` should only be used for the exponent.